tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/ImportContextImpl.java

```diff
@@ -167,6 +167,7 @@ public String generateImports() {
 		for ( String next : imports ) {
 			// don't add automatically "imported" stuff
 			if ( !isAutoImported( next ) ) {
+				next = next.replaceAll( ">+$", "" );
```

Comment:
I _highly_ doubt that this is the correct fix.

The correct solution would be to track down when and why an incorrectly-formed type is being added to the list of things to import.

Reply:
Yes, you are absolutely right. I've simply overlooked the fact that the original tests will  fail if in org.hibernate.jpamodelgen.ImportContextImpl#isAutoImported is removed
inSamePackage( next )
I'll close this PR